src/ChannelManager.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <unordered_map>
+#include <set>
+
+#include "StringUtils.h"
+
+
+class cClientHandle;
+class cCriticalSection;
+class cPluginLua;
+class cChannelCallback;
+
+
+// tolua_begin
+class cChannelManager
+{
+
+public:
+	typedef SharedPtr<cChannelCallback> cChannelCallbackRef;
+	typedef std::unordered_map<AString, cChannelCallbackRef> CallbackMap;
+
+	void AddClientToChannel      (cClientHandle & a_Client, const AString & a_Channel);
+	void AddClientToChannels     (cClientHandle & a_Client, const AString & a_Channels);
+	void RemoveClientFromChannel (cClientHandle & a_Client, const AString & a_Channel);
+	void RemoveClientFromChannels(cClientHandle & a_Client, const AString & a_Channels);
+	bool ClientHasChannel        (cClientHandle & a_Client, const AString & a_Channel);
```

Comment:
Using one channel per village sounds like a really bad idea. If it's a mod / bot that you're writing, you should use a single channel and multiplex the data on it instead.

Don't forget that channel names have a limited length, 16 ascii chars if I remember correctly; there's no way to stuff a plugin name and a unique village identification into that.